crates/re_ui/examples/re_ui_example.rs

```diff
@@ -600,3 +645,788 @@ impl egui_tiles::Behavior<Tab> for MyTileTreeBehavior {
         }
     }
 }
+
+// ==============================================================================
+// DRAG AND DROP DEMO
+
+mod drag_and_drop {
+    use std::collections::HashSet;
+
+    #[derive(Hash, Clone, Copy, PartialEq, Eq)]
+    struct ItemId(u32);
+
+    pub struct ExampleDragAndDrop {
+        items: Vec<(ItemId, String)>,
+
+        /// currently selected items
+        selected_items: HashSet<ItemId>,
+    }
+
+    impl Default for ExampleDragAndDrop {
+        fn default() -> Self {
+            Self {
+                items: (0..10).map(|i| (ItemId(i), format!("Item {i}"))).collect(),
+                selected_items: HashSet::new(),
+            }
+        }
+    }
+
+    impl ExampleDragAndDrop {
+        pub fn ui(&mut self, re_ui: &crate::ReUi, ui: &mut egui::Ui) {
+            let mut source_item_pos = None;
+            let mut target_item_pos = None;
+
+            for (i, (item_id, label)) in self.items.iter_mut().enumerate() {
```

Comment:
`iter_mut` threw me for an extra loop, looking for where the item_id and/or label were being mutated

Reply:
I swear I _was_ mutating something at some point 🤷🏻